bevm/service_test.go

```diff
@@ -1,11 +1,280 @@
 package bevm
 
 import (
+	"encoding/hex"
+	"encoding/json"
+	"math/big"
+	"strconv"
+	"strings"
 	"testing"
 
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+	"go.dedis.ch/cothority/v3/byzcoin"
+	"go.dedis.ch/kyber/v3/suites"
+	"go.dedis.ch/onet/v3"
 	"go.dedis.ch/onet/v3/log"
 )
 
+var tSuite = suites.MustFind("Ed25519")
+
 func TestMain(m *testing.M) {
 	log.MainTest(m)
 }
+
+func setupTest() (*onet.LocalTest, *onet.Roster, *Client) {
+	local := onet.NewTCPTest(tSuite)
+
+	// Generate 1 host, don't connect, process messages, don't register the
+	// tree or entitylist
+	_, ro, _ := local.GenTree(1, false)
+
+	client := &Client{onetClient: local.NewClient(ServiceName)}
+
+	// Do not check leaking goroutines (Ethereum leaves goroutines running...)
+	local.Check = onet.CheckNone
```

Comment:
Now that I think of it - would this work with something like
```
log.AddUserUninterestingGoroutine("name of ethereum go-routine")
```
? Then it would still check for leftover onet go-routines...

Reply:
Indeed. That code was written before I implemented `log.AddUserUninterestingGoroutine()`...

Reply:
Actually, it is already done in the `service.init()`, so I can remove this.